src/reducers/fileTree.tsx

```diff
@@ -172,6 +207,20 @@ const reducer: Reducer<FileTreeState, ActionType<typeof actions>> = (
         tree,
       };
     }
+    case getType(actions.buildTreePathList): {
+      const { versionId } = action.payload;
+      const { forVersionId, tree } = state;
+
+      if (!versionId || forVersionId !== versionId || !tree) {
```

Comment:
It's not a good idea to depend on `state.tree` because it puts the burden of control on the component. Currently, it's not in full control otherwise you wouldn't need to log a warning and bail. 

I suggest performing this reducer logic at the same time you [build the tree](https://github.com/mozilla/addons-code-manager/blob/5d836b97265d8b49368a0c12eba8ac0646260f3f/src/reducers/fileTree.tsx#L200). The only downside might be performance but I don't think that will be an issue. If it becomes an issue, we could split the tree building reducer into smaller pieces.

If you really think it's a better idea to depend on `state.tree` then you should pass it through the action payload instead. With that approach, the component would be fully in control.

Reply:
My original intention was to build this at the same time as the tree, but I thought you requested I not do that and make it distinct actions. I now realize that that was probably about storing the linter messages, and not simply the paths. I'll make that change which will change a lot of this patch.

Reply:
> My original intention was to build this at the same time as the tree, but I thought you requested I not do that and make it distinct actions

Ah, sorry for the confusion. I was just requesting to start with a smaller patch (the first one for just files, not linter messages) and improve it incrementally. That makes each patch easier to review.